### PR TITLE
FF2 2.0 fixes (2)

### PR DESCRIPTION
**Pull request by naydef**
Additions, bug fixes and more

### CODE REVIEW

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 1]
```diff
@@ -1902,8 +1900,8 @@ public Action:StartResponseTimer(Handle:timer)
 	decl String:sound[PLATFORM_MAX_PATH];
 	if(FindSound("begin", sound, sizeof(sound)))
 	{
-		EmitSoundToAllExcept(FF2SOUND_MUTEVOICE, sound, Boss[0]);
```
Comment (winstliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp:1905]: Why were these removed?

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 2]
```diff
@@ -1956,7 +1954,7 @@ public Action:Timer_PrepareBGM(Handle:timer, any:userid)
 	{
 		return Plugin_Stop;
 	}
-
+
```
Comment (winstliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: :fire:

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 3]
```diff
@@ -2058,11 +2061,12 @@ PlayBGM(client)
 				SetArrayCell(timeArray, index, time2);
 			}
 		}
-
-		Format(temp, sizeof(temp), "sound/%s", GetArrayCell(musicArray, index));
+
```
Comment (winstliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Get rid of tabs on this line

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 4]
```diff
@@ -2094,7 +2098,7 @@ StartMusic(client=0)
 	{
 		StopMusic(client);
 		playBGM[client]=true;
-		CreateTimer(0.1, Timer_PrepareBGM, GetClientUserId(client), TIMER_FLAG_NO_MAPCHANGE);
+		CreateTimer(0.2, Timer_PrepareBGM, GetClientUserId(client), TIMER_FLAG_NO_MAPCHANGE);
```
Comment (winstliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Why was this changed?

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 5]
```diff
@@ -2165,11 +2169,14 @@ stock EmitSoundToAllExcept(soundFlags, const String:sample[], entity=SOUND_FROM_
 	{
 		return;
 	}
-
-	EmitSound(clients, total, sample, entity, channel, level, flags, volume, pitch, speakerentity, origin, dir, updatePos, soundtime);
+	
+	for(int i=0; i<total; i++)
+	{
+		EmitSoundToClient(clients[i], sample, entity, channel, level, flags, volume, pitch, speakerentity, origin, dir, updatePos, soundtime);
```
Comment (winstliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Any reason this was changed to EmitSoundToClient?

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 6]
```diff
@@ -2178,30 +2185,44 @@ stock bool:CheckSoundFlags(client, soundFlags)
...
 	if(IsFakeClient(client))
 	{
-		return true;
+		return false;
 	}
 
 	if(muteSound[client] & soundFlags)
 	{
-		return false;
+		return true;
```
Comment (winstliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: this should be kept as return false

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 7]
```diff
@@ -2178,30 +2185,44 @@ stock bool:CheckSoundFlags(client, soundFlags)
...
 	}
-	return true;
+	return false;
```
Comment (winstliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Same here, it should be return true

**Review by winstliu — CHANGES_REQUESTED** [review]

**Comment by winstliu** [comment]
CheckSoundFlags should return true if the sound can be played, not if the client has the sound flag.  Maybe the function needs a better name to reflect that.

**Comment by naydef** [comment]
Yea, probably needs better name and not changing functions

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 4 above)
Comment (naydef) [reply addons/sourcemod/scripting/freak_fortress_2.sp]: Avoid bgm not playing again, but probably this is not the cause

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 5 above)
Comment (naydef) [reply addons/sourcemod/scripting/freak_fortress_2.sp]: Was testing with different functions, will revert that after I test

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 6 above)
Comment (naydef) [reply addons/sourcemod/scripting/freak_fortress_2.sp]: Changing everything

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 1 above)
Comment (naydef) [reply addons/sourcemod/scripting/freak_fortress_2.sp:1905]: This one I specifically left because it's the intro sound, with Boss[0] argument the sound cannot be heard. Without it, it can be heard by all players. I already explained this in steam messages i sent you

**Comment by naydef** [comment]
All changes are already tested

File: addons/sourcemod/scripting/include/freak_fortress_2.inc [hunk 8]
```diff
@@ -398,6 +403,33 @@ native FF2_SetClientGlow(client, Float:time1, Float:time2=-1.0);
...
+ * @noreturn
+ */
+native FF2_ClearSoundFlags(client, flags);
+
+/**
+ * Check sound flags for specified client
+ *
+ * @param client    Client's index
+ * @param flags     Flags which to check
+ * @noreturn
```
Comment (winstliu) [inline addons/sourcemod/scripting/include/freak_fortress_2.inc]: This is incorrect

**Review by winstliu — APPROVED** [review]
